Source/d_loop.c

```diff
@@ -219,11 +220,21 @@ void NetUpdate (void)
     NET_SV_Run();
 
     // check time
+    if (fastdemo || clock_rate != 100 || demowarp >= 0 || demoskip)
```

Comment:
Sorry, but this is getting too messy with way too many exceptions.

My preferred solution:
1. `I_GetTime()` is called unconditionally everywhere (as before).
2. `GetAdjustedTime()` is demoted to one possible `I_GetTime()` implementation, just as `I_GetTime_RealTime()`, `I_GetTime_Scaled()` and `I_GetTime_FastDemo()`.
3. `I_GetTime()` gets set to point to `GetAdjustedTime()` in `I_Init()` if and only if a `new_sync` network game is running.

Do you think this is possible?

Alternatively:
2. `GetAdjustedTime()` seems to be identical to `I_GetTime_RealTime()` if `offsetms == 0`. Make `GetAdjustedTime()` the new `I_GetTime_RealTime()` but make sure that `offsetms = 0` except a `new_sync` network game is running. I still prefer the solution outlined above, though.

As an aside: I guess interpolation can get fixed in Woof! (and Crispy) network games if the formula for calculating the value of `fractionaltic` takes `GetAdjustedTime()` into account instead of using straight `I_GetTimeMS()`.

Reply:
> As an aside: I guess interpolation can get fixed in Woof! (and Crispy) network games if the formula for calculating the value of fractionaltic takes GetAdjustedTime() into account instead of using straight I_GetTimeMS().

I tried this, doesn't work. It's better and sometimes the sync is correct, but then it's jitter again. Maybe the formula needs some changes, I'm not sure.

> GetAdjustedTime() is demoted to one possible I_GetTime() implementation

I will try this.

Reply:
> I_GetTime() gets set to point to GetAdjustedTime() in I_Init() if and only if a new_sync network game is running.

`I_GetTime()` is called in places where `offsetms` correction is not needed. `GetAdjustedTime()` is called only in `NetUpdate()`.